extensions/chathistory.md

```diff
@@ -20,13 +28,13 @@ The server as mentioned in this document may refer to either an IRC server or an
 ## Implementation
 The `chathistory` extension uses the [chathistory][batch/chathistory] batch type and introduces a client command, `chathistory`.
 
-To fully support this extension, clients MUST support the [`batch`][batch], [`server-time`][server-time] and [`message-tags`][message-tags] capabilities.
+Full support for this extension requires support for the [`batch`][batch], [`server-time`][server-time] and [`message-tags`][message-tags] capabilities. However, limited functionality is available without support for these CAPs. Servers SHOULD NOT enforce that clients support all related capabilities before using the `chathistory` extension.
```

Comment:
What's the rationale behind this change? Having an implementation that doesn't use these only makes it possible to have half completed implementations when we should be pushing for complete implementations. With the issues of deduplication that has been discussed for this spec so far, the lack of say message-tags would make this impossible.

Reply:
@kylef thought that `chathistory` wouldn't be usable as a replacement for `znc.in/playback`, because `znc.in/playback` is usable without support for those caps. However, I think it's very advantageous for `chathistory` to have feature parity with `znc.in/playback` (and in fact you can express any `*playback` message as either a CHATHISTORY AFTER or a CHATHISTORY BETWEEN query and get the same results --- maybe you'll get a duplicated message but it's not the end of the world).

If you see the pseudocode I added here: https://github.com/prawnsalad/ircv3-specifications/pull/2 it offers the possibility of using CHATHISTORY without full deduplication support.

Reply:
Jess also mentioned this: https://git.causal.agency/pounce/about/pounce.1#Extensions

Basically, right now, the only option is to have the server guess how much history you're missing and play it back. That's one endpoint of a spectrum, and the other endpoint is CHATHISTORY with full deduplication support. But there is an appetite for intermediate points on that spectrum, like `znc.in/playback` and `casual.agency/consumer`, for use by clients that don't want to build the full infrastructure needed to support deduplication. My feeling is that CHATHISTORY will cover those cases if we let it, i.e., by removing the MUST language for dependent caps as done here.